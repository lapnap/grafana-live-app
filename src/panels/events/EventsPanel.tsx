// Libraries
import React, {PureComponent} from 'react';

// Types
import {EventsOptions} from './types';
import {PanelProps} from '@grafana/ui';
import {CanvasElement} from './CanvasElement';

export interface Props extends PanelProps<EventsOptions> {}

interface State {
  html: string;
}

export class EventsPanel extends PureComponent<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = {
      html: 'xxxx',
    };
  }

  counter = 0;

  draw = (ctx: CanvasRenderingContext2D, width: number, height: number) => {
    if (!ctx) {
      console.warn('WHY?', ctx, width, height);
      return;
    }

    // Clear the background
    ctx.fillStyle = '#FF0';
    ctx.fillRect(0, 0, width, height);

    ctx.lineWidth = 1;
    ctx.textBaseline = 'middle';
    ctx.font = '14px "Open Sans", Helvetica, Arial, sans-serif';
    ctx.fillStyle = '#000';
    ctx.fillText(`${this.counter}: (${Math.floor(width)}x${Math.floor(height)})`, 10, height / 2);
    this.counter++;
  };

  // viewbox: <min-x> <min-y> <width> <height>
  render() {
    return (
      <div>
        TODO! show events.
        <br />
        <CanvasElement {...this.props} width="100%" height={25} draw={this.draw} />
      </div>
    );
  }
}
