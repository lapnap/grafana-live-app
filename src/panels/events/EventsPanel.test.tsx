import React from 'react';
import {shallow} from 'enzyme';
import {EventsPanel, Props} from './EventsPanel';

const setup = (propOverrides?: object) => {
  const props: Props = {
    height: 300,
    width: 300,
  } as Props; // partial

  Object.assign(props, propOverrides);

  const wrapper = shallow(<EventsPanel {...props} />);
  const instance = wrapper.instance() as EventsPanel;

  return {
    instance,
    wrapper,
  };
};

describe('Render Events Panel with basic options', () => {
  it('should render', () => {
    const {wrapper} = setup();
    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });
});
