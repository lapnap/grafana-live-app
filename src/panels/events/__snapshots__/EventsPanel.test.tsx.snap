// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Events Panel with basic options should render 1`] = `
<div>
  TODO! show events.
</div>
`;
